liblib/CMakeLists.txt

```diff
@@ -4,194 +4,226 @@
 # University Corporation for Atmospheric Research/Unidata.
 
 # See netcdf-c/COPYRIGHT file for more info.
-SET(liblib_LIBS dispatch netcdf3)
 
 #####
 # Add target objects/modules based on options.
 #####
-
-IF(USE_NETCDF4)
-  SET(liblib_LIBS ${liblib_LIBS} netcdf4)
-ENDIF()
-IF(USE_HDF5)
-  SET(liblib_LIBS ${liblib_LIBS} netcdfhdf5)
-ENDIF()
-
-IF(USE_PNETCDF)
-  SET(liblib_LIBS ${liblib_LIBS} netcdfp)
-ENDIF()
-
-IF(USE_HDF4)
-  SET(liblib_LIBS ${liblib_LIBS} netcdfhdf4)
-ENDIF()
-
-IF(ENABLE_DAP2)
-  SET(liblib_LIBS ${liblib_LIBS} oc2 dap2)
-ENDIF()
-
-IF(ENABLE_DAP4)
-  SET(liblib_LIBS ${liblib_LIBS} dap4)
-  SET(liblib_LIBS ${liblib_LIBS} ncxml)
-ENDIF()
-
-IF(ENABLE_NCZARR)
-  SET(liblib_LIBS ${liblib_LIBS} nczarr)
-ENDIF()
-
-IF(ENABLE_S3_INTERNAL)
-  SET(liblib_LIBS ${liblib_LIBS} ncxml)
-ENDIF()
-
-IF(ENABLE_PLUGINS)
-  SET(liblib_LIBS ${liblib_LIBS} ncpoco)
-ENDIF()
-
-FOREACH(LIBS ${liblib_LIBS})
-  SET(LARGS ${LARGS} $<TARGET_OBJECTS:${LIBS}>)
-ENDFOREACH()
-
-ADD_LIBRARY(netcdf nc_initialize.c ${LARGS} )
-
-IF(STATUS_PARALLEL)
+target_sources(netcdf
+  PRIVATE
+    $<TARGET_OBJECTS:netcdf3>
+    $<TARGET_OBJECTS:dispatch>
+)
```

Comment:
Can use link_libraries with object targets instead

Reply:
I had originally made all of these be `target_link_library`, and then changed them to this. I don't remember why... I'll try to change them back to `target_link_library`

Reply:
@LecrisUT I changed these to all `target_link_libraries` and now the one-off tests fail with this error

> CMake Error: install(EXPORT "netCDFTargets" ...) includes target "netcdf" which requires target "netcdf3" that is not in any export set.

for many/all of the linked libraries. Is there any harm in adding them to the export set that you're aware of, or should I dig a little deeper for a different solution?

Reply:
Oh yeah, I remember that issue, but I don't think I ever managed to work it out. I think you can abuse the generator expression to make the linkage only on build-time, but then I don't know if the generated library/executable is properly packaged. I also remember having more luck with the `INTERFACE` design in the upstream documentation I linked above.

Another thing you can try is to export that object target and look into the `netCDFTargets` and see how exactly each of these are linked and what they each define. Iirc the issue given there is that there's an effective `target_link_libraries(INTERFACE)` between those 2, but I don't see why that would be. The targets file should confirm this at least.

Reply:
It seems most of the issues have been figured out.